packages/client/modules/userDashboard/components/UserTasksHeader/UserTasksHeader.tsx

```diff
@@ -42,37 +65,83 @@ const StyledCheckbox = styled(Checkbox)({
 
 const UserTasksHeaderDashSectionControls = styled(DashSectionControls)({
   justifyContent: 'flex-start',
+  flexWrap: 'wrap',
+  width: '100%'
 })
 
 interface Props {
   viewer: UserTasksHeader_viewer
 }
 
 const UserTasksHeader = (props: Props) => {
-  const atmosphere = useAtmosphere()
+  const {history} = useRouter()
   const {viewer} = props
-  const {menuPortal, togglePortal, originRef, menuProps} = useMenu(MenuPosition.UPPER_RIGHT, {
+  const {
+    menuPortal: teamFilterMenuPortal,
+    togglePortal: teamFilterTogglePortal,
+    originRef: teamFilterOriginRef,
+    menuProps: teamFilterMenuProps
+  } = useMenu(MenuPosition.UPPER_RIGHT, {
+    isDropdown: true
+  })
+  const {
+    menuPortal: teamMemberFilterMenuPortal,
+    togglePortal: teamMemberFilterTogglePortal,
+    originRef: teamMemberFilterOriginRef,
+    menuProps: teamMemberFilterMenuProps
+  } = useMenu(MenuPosition.UPPER_RIGHT, {
     isDropdown: true
   })
-  const {teamFilter, showArchivedTasksCheckbox} = viewer
-  const teamFilterName = (teamFilter && teamFilter.name) || 'All teams'
+  const {teams} = viewer
+  const teamMembers = teams.map(({teamMembers}) => teamMembers).flat()
+  const users = [...new Set(teamMembers.map(({user}) => user).flat())]
```

Comment:
`Set` only works to dedupe things are are shallowly equal (e.g. `{} !== {}`). i don't think relay guarantees that it reuses the same user object, so you'll have to dedupe on ID instead of just object reference.